source/future.tex

```diff
@@ -1459,6 +1459,123 @@
 \end{note}
 \end{itemdescr}
 
+\rSec1[depr.tuple]{Tuple}
+
+Header \libheaderref{tuple}:
+
+\begin{codeblock}
+namespace std {
+  // \ref{tuple.helper}, tuple helper classes
+  template<class T> class tuple_size<volatile T>;
+  template<class T> class tuple_size<const volatile T>;
+
+  template<size_t I, class T> class tuple_element<I, volatile T>;
+  template<size_t I, class T> class tuple_element<I, const volatile T>;
+}
+\end{codeblock}
+
+\begin{itemdecl}
+template<class T> class tuple_size<volatile T>;
+template<class T> class tuple_size<const volatile T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{TS} denote \tcode{tuple_size<T>} of the cv-unqualified type \tcode{T}.
+If the expression \tcode{TS::value} is well-formed
+when treated as an unevaluated operand,
+then each of the two templates shall satisfy
+the \tcode{TransformationTrait} requirements with a base characteristic of
+\begin{codeblock}
+integral_constant<size_t, TS::value>
+\end{codeblock}
+Otherwise, they shall have no member \tcode{value}.
+
+\pnum
+Access checking is performed as if
+in a context unrelated to \tcode{TS} and \tcode{T}.
+Only the validity of the immediate context of the expression is considered.
+
+\pnum
+In addition to being available via inclusion of the \libheaderref{tuple} header,
+the two templates are available when any of the headers
+\libheaderref{array},
+\libheaderref{ranges}, or
+\libheaderref{utility}
+are included.
+\end{itemdescr}
+
+\begin{itemdecl}
+template<size_t I, class T> class tuple_element<I, volatile T>;
+template<size_t I, class T> class tuple_element<I, const volatile T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{TE} denote \tcode{tuple_element_t<I, T>}
+of the cv-unqualified type \tcode{T}.
+Then each of the two templates shall satisfy
+the \tcode{TransformationTrait} requirements
+with a member typedef type that names the following type:
```

Comment:
Bad wording - how can a member typedef satisfy requirements for a template?

Reply:
This is a usual pattern when using these type traits. The idea is that a "Transformation Trait" requires a member typedef, and that member typedef is then required to name "the following type".